aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -117,9 +120,39 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (exceptionThrown) {
         io().deleteFile(newMetadataLocation);
       }
+
+      unlock();
+    }
+  }
+
+  private void acquireLock() {
+    try {
+      boolean lockAcquired = lockManager.tryLock(lockId(databaseName, tableName),
+          awsProperties.glueCatalogLockExpireMs(), awsProperties.glueCatalogLockWaitMs());
+      if (!lockAcquired) {
+        throw new CommitFailedException("Timed out after %s ms waiting for lock on %s.%s",
+            awsProperties.glueCatalogLockWaitMs(), databaseName, tableName);
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new CommitFailedException(e, "Thread interrupted during commit");
+    } catch (Exception e) {
+      throw new CommitFailedException(e, "Unexpected exception when locking %s.%s", databaseName, tableName);
     }
   }
 
+  private void unlock() {
+    try {
+      lockManager.unlock(lockId(databaseName, tableName));
+    } catch (Exception e) {
+      LOG.error("Unexpected exception when unlocking {}.{}, manual unlock might be needed", databaseName, tableName, e);
```

Comment:
This should not swallow the exception, it should re-throw it.

I also think that there is no need to catch `Exception` here because there are no checked exceptions thrown by `unlock`. `RuntimeException` should be fine.